backend/src/main/java/meowhub/backend/security/SecurityConfig.java

```diff
@@ -53,40 +59,56 @@ SecurityFilterChain defaultSecurityFilterChain(HttpSecurity http, AuthEntryPoint
     }
 
     @Bean
-    public CommandLineRunner initData(RoleRepository roleRepository, UserRepository userRepository, PasswordEncoder passwordEncoder) {
+    public CommandLineRunner initData(RoleRepository roleRepository, UserRepository userRepository, PasswordEncoder passwordEncoder, PrivacySettingRepository privacySettingRepository, GenderRepository genderRepository) {
         return args -> {
-            Role userRole = roleRepository.findByRoleName(ApplicationRole.ROLE_USER)
-                    .orElseGet(() -> roleRepository.save(new Role(ApplicationRole.ROLE_USER)));
-            Role adminRole = roleRepository.findByRoleName(ApplicationRole.ROLE_ADMIN)
-                    .orElseGet(() -> roleRepository.save(new Role(ApplicationRole.ROLE_ADMIN)));
+            Role userRole = roleRepository.findByCode(Roles.ROLE_USER.name())
+                    .orElseGet(() -> roleRepository.save(new Role(Roles.ROLE_USER)));
+            Role adminRole = roleRepository.findByCode(Roles.ROLE_ADMIN.name())
+                    .orElseGet(() -> roleRepository.save(new Role(Roles.ROLE_ADMIN)));
+
+            PrivacySetting publicSetting = privacySettingRepository.findByCode(PrivacySettings.PUBLIC.name())
+                    .orElseGet(() -> privacySettingRepository.save(new PrivacySetting(PrivacySettings.PUBLIC)));
+
+            Gender female = genderRepository.findByCode(Genders.FEMALE.name())
+                    .orElseGet(() -> genderRepository.save(new Gender(Genders.FEMALE)));
 
             if (!userRepository.existsByLogin("user1")) {
-                User user1 = new User("user1", "user1@example.com", passwordEncoder.encode("password1"));
+                User user1 = new User();
+                user1.setLogin("user1");
+                user1.setPassword(passwordEncoder.encode("password1"));
+                user1.setEmail("user1@example.com");
                 user1.setName("Jan");
                 user1.setSurname("Kos");
+                user1.setSalt("salt");
```

Comment:
i am aware that this approach to hardcode salt is disgusting, but currently we don't have a working mechanism for it yet. Also, this task was meant only to create models with reverse database engineering and adjust the project so it can run with then smoothly.

The salt implementation will be provided in the future, it might be after delivering certain API's like posts API or profiles API though, as they have higher priority now.